common/djangoapps/third_party_auth/saml.py

```diff
@@ -41,10 +42,18 @@ def auth_url(self):
         if not self._config.enabled:
             log.error('SAML authentication is not enabled')
             raise Http404
+
         # TODO: remove this check once the fix is merged upstream:
         # https://github.com/omab/python-social-auth/pull/821
         if 'idp' not in self.strategy.request_data():
             raise AuthMissingParameter(self, 'idp')
+
+        from .models import SAMLProviderConfig
+        provider = SAMLProviderConfig.current(self.strategy.request_data()['idp'])
+        if not provider.enabled_for_current_site:
+            log.error('The SAML authentication provider is disabled for this site')
+            raise Http404
```

Comment:
Looks like this if clause is not [covered](https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/25582/Diff_Coverage_Report/). Any chance you can add coverage?